lib/msf/core/post/file.rb

```diff
@@ -837,7 +837,7 @@ def _read_file_meterpreter(file_name)
     data
   rescue EOFError
     # Sometimes fd isn't marked EOF in time?
-    ''
+    data
```

Comment:
I'm not sure why `''` was returned previously; but it looks like the previous developer hit the same issue I spotted on CI - https://github.com/rapid7/metasploit-framework/commit/5cafd8908ec683fef3a2fd1de3f564583ae9e65f

Note that treating an EOFError as returning the file content is also the pattern used in  other parts of the codebase:

https://github.com/rapid7/metasploit-framework/commit/e200f43183c8b804949ea3d9792740b3d92ada8b#diff-82002a8c61d75afcc08e77592faaa8c0c70f1e2a1e89887b5d1874bc312c6ba8

Reply:
Or this could be:

```ruby
data = ''
data << fd.read
data << fd.read until fd.eof?
```